cleanup: address nlreturn linter issues

Rakshith-R opened the pull request:
`nlreturn` linter requires a new line before return and branch statements except when the return is alone
inside a statement group (such as an if statement) to increase code clarity. This commit addresses such issues.

Updates: #1586

Signed-off-by: Rakshith R <rar@redhat.com>
<details>
<summary>

Expand to see `nlreturn` errors fixed by this pr</summary>

```bash

internal/journal/omap.go:86:2: return with no blank line before (nlreturn)
	return results, nil
	^
internal/journal/omap.go:81:3: return with no blank line before (nlreturn)
		return nil, err
		^
internal/journal/omap.go:79:4: return with no blank line before (nlreturn)
			return nil, util.JoinErrors(util.ErrKeyNotFound, err)
			^
internal/journal/voljournal.go:470:5: continue with no blank line before (nlreturn)
				continue
				^
internal/cephfs/volumemounter.go:126:4: break with no blank line before (nlreturn)
			break
			^
e2e/cephfs.go:327:7: break with no blank line before (nlreturn)
						break
						^
e2e/rbd.go:743:7: break with no blank line before (nlreturn)
						break
						^
internal/rbd/rbd_util.go:606:4: continue with no blank line before (nlreturn)
			continue
			^
internal/rbd/rbd_util_test.go:126:4: continue with no blank line before (nlreturn)
			continue
```
</details>

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

Rakshith-R commented:
@mergifyio rebase

yati1998 commented:
@Rakshith-R is there anything else to add or you can convert this PR from draft.

Rakshith-R commented:
> @Rakshith-R is there anything else to add or you can convert this PR from draft.

I was planning to rerun tests and clear all the nlreturns error just before golangci is upgraded (since other prs will just introduce more of it, if golangci is  not upgraded). 

I can update this pr when golangci update pr comes up or you can take it up too. Let me know what works for you :slightly_smiling_face:

yati1998 commented:
> > @Rakshith-R is there anything else to add or you can convert this PR from draft.
> 
> I was planning to rerun tests and clear all the nlreturns error just before golangci is upgraded (since other prs will just introduce more of it, if golangci is not upgraded).
> 
> I can update this pr when golangci update pr comes up or you can take it up too. Let me know what works for you

I guess you can add changes according to the current code. Cause we can't merge the update golangci version PR before this. We can handle incoming issues later. So no worries.

Madhu-1 commented:
As the change is only to add new lines. skipped the E2E as static check testing is enough.

Rakshith-R commented:
This pr only adds new lines before return/break/continue except incases where its the only statement. 

:sweat_smile:  Used auto fix from https://github.com/ssgreg/nlreturn (+ alot of hacks)

and tested with golanci 1.41.1 nlreturn 
```
cephcsi image settings: quay.io/cephcsi/cephcsi version canary
docker run --rm -v /home/rakshith/workspace/src/github.com/ceph/ceph-csi:/go/src/github.com/ceph/ceph-csi:z quay.io/cephcsi/cephcsi:test make go-lint GIT_SINCE=origin/devel REBASE=0 CONTAINERIZED=yes
cephcsi image settings: quay.io/cephcsi/cephcsi version canary
sed "s/@@CEPH_VERSION@@/octopus/g" < scripts/golangci.yml.in > scripts/golangci.yml
./scripts/lint-go.sh
level=info msg="[config_reader] Used config file scripts/golangci.yml"
level=info msg="[lintersdb] Active 1 linters: [nlreturn]"
level=info msg="[loader] Using build tags: [octopus]"
level=info msg="[loader] Go packages loading at mode 7 (compiled_files|files|name) took 9.044022435s"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 21.19949ms"
level=info msg="[linters context/goanalysis] analyzers took 5.29546ms with top 10 stages: nlreturn: 5.29546ms"
level=info msg="[runner] processing took 3.581µs with stages: nolint: 486ns, max_same_issues: 481ns, skip_dirs: 473ns, filename_unadjuster: 236ns, max_from_linter: 234ns, cgo: 202ns, path_shortener: 179ns, uniq_by_line: 144ns, source_code: 142ns, diff: 140ns, skip_files: 131ns, path_prettifier: 127ns, identifier_marker: 127ns, autogenerated_exclude: 115ns, exclude: 65ns, sort_results: 64ns, max_per_file_from_linter: 62ns, path_prefixer: 62ns, exclude-rules: 57ns, severity-rules: 54ns"
level=info msg="[runner] linters took 32.784078ms with stages: nlreturn: 32.722735ms"
level=info msg="File cache stats: 0 entries of total size 0B"
level=info msg="Memory: 93 samples, avg is 71.6MB, max is 72.5MB"
level=info msg="Execution took 9.102330914s"
```

Rakshith-R commented:
@humblec @nixpanic , PTAL

nixpanic commented:
@Rakshith-R, needs a manual rebase unfortunately

yati1998 commented:
@Rakshith-R please try to fix this by today. It's blocker for #2131

Rakshith-R committed:
cleanup: resolve nlreturn linter issues

nlreturn linter requires a new line before return
and branch statements except when the return is alone
inside a statement group (such as an if statement) to
increase code clarity. This commit addresses such issues.

Updates: #1586

Signed-off-by: Rakshith R <rar@redhat.com>
